h2/src/test/org/h2/test/jdbc/TestPreparedStatement.java

```diff
@@ -564,49 +562,6 @@ private void testUUIDAsJavaObject(Connection conn) throws SQLException {
         stat.execute("drop table test_uuid");
     }
 
-    /**
-     * A trigger that creates a sequence value.
-     */
-    public static class SequenceTrigger implements Trigger {
-
-        @Override
-        public void fire(Connection conn, Object[] oldRow, Object[] newRow)
-                throws SQLException {
-            conn.setAutoCommit(false);
-            conn.createStatement().execute("call next value for seq");
-        }
-
-        @Override
-        public void init(Connection conn, String schemaName,
-                String triggerName, String tableName, boolean before, int type) {
-            // ignore
-        }
-
-    }
-
-    private void testScopedGeneratedKey(Connection conn) throws SQLException {
-        Statement stat = conn.createStatement();
-        stat.execute("create table test(id identity)");
-        stat.execute("create sequence seq start with 1000");
-        stat.execute("create trigger test_ins after insert on test call \"" +
-                SequenceTrigger.class.getName() + "\"");
-        stat.execute("insert into test values(null)", Statement.RETURN_GENERATED_KEYS);
-        ResultSet rs = stat.getGeneratedKeys();
-        rs.next();
-        // Generated key
-        assertEquals(1, rs.getLong(1));
-        stat.execute("insert into test values(100)");
-        rs = stat.getGeneratedKeys();
-        // No generated keys
-        assertFalse(rs.next());
-        // Value from sequence from trigger
-        rs = stat.executeQuery("select scope_identity()");
```

Comment:
I think you broke a valid use-case of scope_identity(), which this test was checking .e.g see
https://stackoverflow.com/a/1920640/6591

Reply:
This test case ensures incorrect (according to our own documentation) behavior:
https://h2database.com/html/functions.html#scope_identity
> Changes within triggers and Java functions are ignored.

This test case shouldn't see anything from the trigger.

In the real SQL Server, `SCOPE_IDENTITY()` actually returns `NULL` in the similar test case with sequence called in the trigger.

Our implementations of `IDENTITY()` and `SCOPE_IDENTITY()` in general aren't fully compatible with anything. In the SQL Server they are about inserted values, not about called sequences.

Reply:
Hmmm. OK. 

In our case, inserting a value will effectively call a sequence to generate a new ID. So the test was really testing for the SQL Server behaviour. 
But in general, SCOPE_IDENTITY is a pain anyhow, so this change is OK.

Reply:
Actually this PR doesn't break existing compatibility SQL Server:
```SQL
-- MS SQL
CREATE TABLE M(ID BIGINT IDENTITY(1, 1) PRIMARY KEY, VALUE INT);
INSERT INTO M VALUES (10);
CREATE TABLE T(ID BIGINT IDENTITY(1, 1) PRIMARY KEY, VALUE INT);
CREATE TRIGGER TT ON M FOR INSERT AS BEGIN INSERT INTO T VALUES (30); END;
INSERT INTO M VALUES (20);
SELECT @@IDENTITY AS [IDENTITY], SCOPE_IDENTITY() AS [SCOPE_IDENTITY];
> 1 2
-- H2 (with and without these changes):
DROP ALL OBJECTS;
CREATE TABLE M(ID IDENTITY PRIMARY KEY, V INT);
INSERT INTO M(V) VALUES 10;
CREATE TABLE T(ID IDENTITY PRIMARY KEY, V INT);
CREATE TRIGGER TT AFTER INSERT ON M AS
'org.h2.api.Trigger create() {
    return new org.h2.api.Trigger() {
        public void fire(Connection conn, Object[] oldRow, Object[] newRow) throws SQLException {
            conn.setAutoCommit(false);
            conn.createStatement().execute("INSERT INTO T(V) VALUES 30");
        }
    };
}';
INSERT INTO M(V) VALUES 20;
SELECT IDENTITY(), SCOPE_IDENTITY();
> 1 2
```

Reply:
But this removed test case tested the wrong behavior that was fixed here.

We have other test for generated keys from the trigger, so I don't see a reason for adjustments in this one.